erpnext/public/js/utils/barcode_scanner.js

```diff
@@ -92,12 +88,11 @@ erpnext.utils.BarcodeScanner = class BarcodeScanner {
 		}
 
 		this.set_selector_trigger_flag(row, data);
-		this.set_item(row, item_code).then(qty => {
-			this.show_scan_message(row.idx, row.item_code, qty);
-		});
-		this.set_serial_no(row, serial_no);
-		this.set_batch_no(row, batch_no);
-		this.set_barcode(row, barcode);
+		const qty = await this.set_item(row, item_code);
+		this.show_scan_message(row.idx, row.item_code, qty);
+		await this.set_serial_no(row, serial_no);
+		await this.set_batch_no(row, batch_no);
+		await this.set_barcode(row, barcode);
```

Comment:
This is practically all sequential code now, so better to use `frappe.run_serially` ?

https://github.com/frappe/frappe/wiki/Developer-Cheatsheet#running-tasks-serially

Reply:
One more thing, all dependent triggers are not ran serially, so if any fields have triggers that call DB and update it later then those changes will likely not be present on row you receive 😅

Reply:
Resolved https://github.com/frappe/erpnext/pull/31058/commits/4ee6ab1b62b8c966492d86a4a2a21d5aed97e0b9